src/fmt/unicode.c

```diff
@@ -147,22 +150,81 @@ int utf8_decode(struct re_printf *pf, const struct pl *pl)
 				u |= ((uint16_t)ch_hex(pl->p[++i])) << 4;
 				u |= ((uint16_t)ch_hex(pl->p[++i])) << 0;
 
-				if (u > 255) {
-					ch  = u>>8;
-					err = pf->vph(&ch, 1, pf->arg);
-					if (err)
-						return err;
+				/* UTF-16 surrogate pair */
+				if (u >= 0xd800 && u <= 0xdbff) {
+					uhi = (u - 0xd800) * 0x400;
+					continue;
 				}
+				else if (u >= 0xdc00 && u <= 0xdfff) {
+					if (uhi < 0)
+						continue;
```

Comment:
this code seems to decode a UTF-16 surrogate pair (first high, then low).

if a corrupt Unicode string is detected, should we just ignore it or stop the decoding
with an error ?

for example this string would be corrupt:

```\\uD834\\uD834```

Reply:
IMO it is best to just ignore invalid code points, so that all the part of the string that is valid gets decoded properly. I think that is the best and most robust behaviour. This patch is written with that in mind.